models/users.js

```diff
@@ -163,57 +163,155 @@ const getSuggestedUsers = async (skill) => {
  * @return {Promise<userModel|Array>}
  */
 const fetchPaginatedUsers = async (query) => {
-  try {
-    // INFO: default user size set to 100
-    // INFO: https://github.com/Real-Dev-Squad/website-backend/pull/873#discussion_r1064229932
-    const size = parseInt(query.size) || 100;
-    const doc = (query.next || query.prev) && (await userModel.doc(query.next || query.prev).get());
+  const isDevMode = query.dev === "true";
+
+  if (isDevMode) {
+    try {
+      const size = parseInt(query.size) || 100;
+      const doc = (query.next || query.prev) && (await userModel.doc(query.next || query.prev).get());
+
+      let dbQuery = userModel.where("roles.archived", "==", false).orderBy("username");
+      const usernameQuery = userModel.where("roles.archived", "==", false).orderBy("username");
+      const firstNameQuery = userModel.where("roles.archived", "==", false).orderBy("first_name");
+      const lastNameQuery = userModel.where("roles.archived", "==", false).orderBy("last_name");
+      let compositeQuery = [usernameQuery, firstNameQuery, lastNameQuery];
```

Comment:
`fetchPaginatedUsers` can be refactored to separate concerns into distinct functions or modules, such as query construction, data retrieval, and error handling. composite query construction can be abstracted into a reusable function to eliminate redundancy and to improve maintainability.

Reply:
Please elaborate.

Reply:
is this a p* for this task?